docs/general/user-defined-functions.rst

```diff
@@ -197,31 +197,48 @@ Optionally, you can provide a schema::
 Supported languages
 ===================
 
-CrateDB currently only supports the UDF language ``javascript``.
+CrateDB currently only supports the ``JavaScript`` user-defined functions.
 
 .. _udf_lang_js:
 
 JavaScript
 ----------
 
-The UDF language ``javascript`` supports the `ECMAScript 5.1`_ standard.
+The user defined function JavaScript is compatible with the `ECMAScript 2019`_
+specification.
 
 .. NOTE::
 
    The JavaScript language is an :ref:`enterprise feature
    <enterprise-features>`.
 
-CrateDB uses the Java built-in JavaScript engine Nashorn_ to interpret and
-execute functions written in JavaScript. The engine is initialized using the
-``--no-java`` option which basically restricts all access to Java APIs from
-within the JavaScript context. CrateDB's engine also does not allow
-non-standard syntax extensions (``--no-syntax-extensions``).
+CrateDB uses the `GraalVM JavaScript`_ engine as a JavaScript (ECMAScript)
+language execution runtime. The `GraalVM JavaScript`_ engine is a Java
+application that works on the stock Java VMs. The interoperability between
+Java code (host language) and JavaScript user-defined functions
+(guest language) is guaranteed thtough the GraalVM `GraalVM Polyglot API`_.
+
+Please note that CrateDB does not use the GraalVM JIT compiler as the
+optimizing compile of the GraalVM JavaScript code as one might expect while
+seeing usage of the GraalVM. The `stock host Java VM JIT compilers`_ still can
+JIT-compile, execute and optimize the GraalVM JavaScript codebase to certain
+extend.
```

Comment:
how about this:

> Please note: CrateDB does not use the GraalVM JIT compiler. However, the `stock host Java VM JIT compilers`_ can
JIT-compile, optimize, and execute the GraalVM JavaScript codebase to a certain extent.